Dockerfile

```diff
@@ -11,12 +11,10 @@ ARG DATABASE_BACKEND=spanner
 
 # cmake is required to build grpcio-sys for Spanner builds
 RUN \
-    # Fetch and load the MySQL public key. We need to install libmysqlclient-dev to build syncstorage-rs
+    # We need to install libmariadb-dev to build syncstorage-rs
     # which wants the mariadb
-    wget -qO- https://repo.mysql.com/RPM-GPG-KEY-mysql-2023 > /etc/apt/trusted.gpg.d/mysql.asc && \
-    echo "deb https://repo.mysql.com/apt/debian/ bullseye mysql-8.0" >> /etc/apt/sources.list && \
     apt-get -q update && \
-    apt-get -q install -y --no-install-recommends libmysqlclient-dev cmake
+    apt-get -q install -y --no-install-recommends libmariadb-dev-compat libmariadb-dev cmake
```

Comment:
nit: `libmariadb-dev` is superfluous w/ `libmariadb-dev-compat` (which depends on it)